core/src/Streamly/Internal/Unicode/Parser.hs

```diff
@@ -263,36 +265,127 @@ hexadecimal = Parser.takeWhile1 isHexDigit (Fold.foldl' step 0)
 signed :: (Num a, Monad m) => Parser Char m a -> Parser Char m a
 signed p = (negate <$> (char '-' *> p)) <|> (char '+' *> p) <|> p
 
+data TE =
+    TE  {-# UNPACK #-}!Int      -- ^ Total number of digits
+        {-# UNPACK #-}!Int      -- ^ Decimal point exsistence
+        {-# UNPACK #-}!Int      -- ^ The sign positive or negative
+        !Integer                -- ^ The coefficient of a scientific number.
+        {-# UNPACK #-}!Int      -- ^ The base-10 exponent of a scientific number.
+
 -- | Parse a 'Double'.
---
--- This parser accepts an optional leading sign character, followed by
--- at most one decimal digit.  The syntax is similar to that accepted by
+-- This parser accepts an optional sign (+ or -) followed by
+-- at least one decimal digit.  The syntax is similar to that accepted by
 -- the 'read' function, with the exception that a trailing @\'.\'@ is
 -- consumed.
+-- This function does not accept string representations of \"NaN\" or
+-- \"Infinity\".
+
+-- >>> import qualified Streamly.Internal.Data.Stream as S
 --
--- === Examples
+-- >>> doubleParser = S.parse double . S.fromList
 --
--- Examples with behaviour identical to 'read', if you feed an empty
--- continuation to the first result:
+-- >>> doubleParser "3"
+-- Right 3.0
 --
--- > IS.parse double (IS.fromList "3")     == 3.0
--- > IS.parse double (IS.fromList "3.1")   == 3.1
--- > IS.parse double (IS.fromList "3e4")   == 30000.0
--- > IS.parse double (IS.fromList "3.1e4") == 31000.0
--- > IS.parse double (IS.fromList "3e")    == 30
+-- >>> doubleParser ".3"
+-- Right 0.3
 --
--- Examples with behaviour identical to 'read':
+-- >>> doubleParser "00.3"
+-- Right 0.3
 --
--- > IS.parse (IS.fromList ".3")    == error "Parse failed"
--- > IS.parse (IS.fromList "e3")    == error "Parse failed"
+-- >>> doubleParser "0.003"
+-- Right 3.0e-3
 --
--- Example of difference from 'read':
+-- >>> doubleParser "1000.003"
+-- Right 1000.003
 --
--- > IS.parse double (IS.fromList "3.foo") == 3.0
+-- >>> doubleParser "+123.345"
+-- Right 123.345
 --
--- This function does not accept string representations of \"NaN\" or
--- \"Infinity\".
+-- >>> doubleParser "-123.345"
+-- Right (-123.345)
+--
+-- >>> doubleParser "0.0.00"
+-- Right 0.0
+--
+-- >>> doubleParser "44.56.67"
+-- Right 44.56
+--
+-- >>> doubleParser "44.56.67-9"
+-- Right 44.56
+--
+-- >>> doubleParser "44-6008"
+-- Right 44.0
+--
+-- >>> doubleParser "44+36008"
+-- Right 44.0
 --
--- /Unimplemented/
-double :: Parser Char m Double
-double = undefined
+-- >>> doubleParser "447u"
+-- Right 447.0
+--
+-- >>> doubleParser "+"
+-- Right 0.0
+--
+-- >>> doubleParser "-"
+-- Right 0.0
+--
+-- >>> doubleParser ""
+-- Right 0.0
+--
+-- >>> doubleParser "+."
+-- Right 0.0
+--
+-- >>> doubleParser "-."
+-- Right 0.0
+--
+-- >>> doubleParser "4."
+-- Right 4.0
+--
+-- >>> doubleParser "a."
+-- Left (ParseError "Invalid double fromat")
+--
```

Comment:
Many of the above tests do not look right. Can we make sure that the results are compliant with the double parser in attoparsec https://hackage.haskell.org/package/attoparsec-0.14.4/docs/Data-Attoparsec-ByteString-Char8.html#g:14 ?